scripts/get_og_previews.py

```diff
@@ -55,29 +60,61 @@ def get_og_preview(url):
     # TODO: Maybe add additional behaviour to check if the image is large enough
     soup = get_page(url)
     image_url = get_og_image(soup)
+
     return image_url
 
 
+def save_image_as_webp(binary_image: bytes, path: Path, file_name: str) -> Path:
+    full_path = path / f"{file_name}.webp"
+    # TODO: Might need to add some validation to make sure the binary is an image
+    # Depends on if Pillow will just write a broken binary instead in those cases
+    img = Image.open(BytesIO(binary_image))
+    img.save(full_path, "webp")
+    return full_path
+
+
+def write_image_to_file(url: str, file_path: Path) -> Path | None:
+    file_name = hashlib.shake_128(url.encode("utf-8")).hexdigest(4)
+    image_path = Path("static") / "previews"
+
+    # Add Mozilla header to prevent getting blocked for scraping
+    r = httpx.get(url, headers={"User-agent": "Mozilla/5.0"}, follow_redirects=True)
+
+    # Site had no actual image in their og_image url, so no point saving it
+    if r.status_code != 200:
+        logger.error(f"Couldn't find any image at {url}")
+        return None
+
+    return save_image_as_webp(r.content, image_path, file_name)
+
+
 def main():
-    with resources_file.open() as f:
+    # Ensure the path for our previews actually exists
+    PREVIEW_PATH.mkdir(parents=True, exist_ok=True)
+
+    with RESOURCES_FILE.open() as f:
         resources = yaml.safe_load(f)
     logger.success("Read in `resources.yml` file.")
 
     for resource in resources:
         logger.info(f"Getting OG preview for {resource['url']}")
 
         try:
-            image = get_og_preview(resource["url"])
+            image_url = get_og_preview(resource["url"])
         except Exception as e:
             logger.error(e)
-            image = None
+            image_url = None
 
-        if image:
+        if image_url:
             # Check if image is valid URL
-            if validators.url(image):
-                resource["og_preview"] = image
-
-    with resources_file.open("w") as f:
+            # TODO: Remove this nesting by offloading the validation elsewhere
```

Comment:
I'll fix this. Sure, we shouldn't nest too far, but at the same time this pretty simple functionality that doesn't justify a new function :)